build(recipe): add recipe for conda-forge

deepcharles opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->
I also added two maintainers for this recipe (@oboulant and myself).

deepcharles committed:
build(recipe): add recipe for conda-forge

deepcharles committed:
build(recipe): add recipe for conda-forge

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

deepcharles committed:
build: remove wheel as dependency

deepcharles committed:
build: remove wheel as dependency

deepcharles commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.20.5, conda-smithy 3.8.5, and conda-forge-pinning 2020.12.02.18.06.47

ogrisel commented:
As the maintainer of the upstream package, you should probably add yourself to the `recipe-maintainers` section of the `meta.yaml` file. This way you will be able to merge your own maintenance PRs on this repo.

deepcharles committed:
build(maintainer): adding myself and oboulant

As maintainers of the upstream package, we would
be able to make updates on our own.

deepcharles committed:
Merge branch 'master' of https://github.com/deepcharles/ruptures-feedstock

deepcharles commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.20.5, conda-smithy 3.8.5, and conda-forge-pinning 2020.12.02.18.06.47

oboulant commented:
I am willing to be listed here 🙂

deepcharles commented:
@oboulant I added you already (in `recipe/meta.yaml`).

oboulant commented:
@deepcharles I know but I just confirmed it. In the docs for the recipe creation they list as an item to post a comment of the maintainers mentioned. Since it is an addition here, I thought it might be required to have consent. So, to clear the doubt I posted the comment :)

oboulant commented:
@conda-forge-admin We have a PR passing all the checks and we were wondering what would be the next steps for the PR to be merged ?
Thanks a lot again !

xhochy commented on recipe/meta.yaml:
```diff
@@ -1,27 +1,34 @@
+  skip: True  # [py<36]
   script: {{ PYTHON }} -m pip install . -vv
-  noarch: python
+  # noarch: python
```
```suggestion
```

xhochy commented on recipe/meta.yaml:
```diff
@@ -1,27 +1,34 @@
 {% set name = "ruptures" %}
-{% set version = "1.0.6" %}
+{% set version = "1.1.2rc1" %}
```
Don't publish release candidates on the main label.

deepcharles replied:
Noted. Then I'll wait a few days to push the 1.1.2 version. In the meantime I have two questions:

- can you merge #8 in which I added myself and @oboulant as maintainers?
- can you point to some documentation for pushing release candidates? I have seen some discussions but nothing "official".

xhochy replied:
Have a look at https://github.com/conda-forge/cfep/blob/master/cfep-05.md

deepcharles replied:
Many thanks!

xhochy reviewed (changes requested):
This is a PR for a release candidate, this should not be merged to the main label

deepcharles committed:
build(meta.yaml): remove commented line

Co-authored-by: Uwe L. Korn <xhochy@users.noreply.github.com>

deepcharles committed:
build(meta.yaml): remove commented line

Co-authored-by: Uwe L. Korn <xhochy@users.noreply.github.com>

deepcharles committed:
build(meta.yaml): add maintainers

@oboulant and myself are maintainers of the upstream package (see [here](https://pypi.org/project/ruptures/))

Commit:
MNT: Re-rendered with conda-build 3.20.5, conda-smithy 3.8.5, and conda-forge-pinning 2020.12.07.18.58.57

github-actions[bot] committed:
build(meta.yaml): add maintainers

automerged PR by conda-forge/automerge-action

rpanai commented:
> @conda-forge-admin We have a PR passing all the checks and we were wondering what would be the next steps for the PR to be merged ?
> Thanks a lot again !

I've been asked to review the PR but as @xhochy said you are trying to push a release candidate. When do you think the production release will be available?

deepcharles commented:
> When do you think the production release will be available?

Next week probably.

deepcharles committed:
build(meta.yml): bump version to 1.1.2

update version number and associated sha256

deepcharles committed:
build(meta.yml): bump version to 1.1.2

update version number and associated sha256

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I was trying to look for recipes to lint for you, but it appears we have a merge conflict.
Please try to merge or rebase with the base branch to resolve this conflict.

Please ping the 'conda-forge/core' team (using the @ notation in a comment) if you believe this is a bug.

deepcharles commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.20.5, conda-smithy 3.8.6, and conda-forge-pinning 2021.01.01.16.59.39

Commit:
MNT: Re-rendered with conda-build 3.20.5, conda-smithy 3.8.6, and conda-forge-pinning 2021.01.01.16.59.39

deepcharles committed:
Merge branch 'master' of https://github.com/deepcharles/ruptures-feedstock

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

deepcharles commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.20.5, conda-smithy 3.8.6, and conda-forge-pinning 2021.01.01.16.59.39

deepcharles commented:
This is now the correct version of `ruptures` (i.e. not a pre-release) to upload to conda-forge.

@rpanai @xhochy The PR can now be merge to master

xhochy commented on recipe/meta.yaml:
```diff
@@ -1,27 +1,33 @@
 
 build:
   number: 0
+  skip: True  # [py<36]
```
This is not needed, we skip Python <3.6 automatically.

deepcharles replied:
thanks for the quick reply. Resolved in c25ae3f.

xhochy commented on recipe/meta.yaml:
```diff
@@ -1,27 +1,33 @@
 requirements:
+  build:
+      # If your project compiles code (such as a C extension) then add the required compilers as separate entries here.
+      # Compilers are named 'c', 'cxx' and 'fortran'.
```
Please remove the comment

deepcharles replied:
resolved in c25ae3f

deepcharles committed:
build: clean useless comments and instructions

- https://github.com/conda-forge/ruptures-feedstock/pull/7#discussion_r550797563
- https://github.com/conda-forge/ruptures-feedstock/pull/7#discussion_r550797638

deepcharles commented:
@conda-forge-admin, please rerender

xhochy commented:
@rpan PTAL. It's fine for but I would wait a bit for you before the merge.

oboulant reviewed (approved):
LGTM
Thanks !

oboulant commented:
@rpanai Did you have any chances to have a look at this PR ? If yes, could you approve so we can move forward with the conda release of the v1.1.2 of the `ruptures` package ? 
Many thanks for the time and energy !

rpanai commented:
> @rpanai Did you have any chances to have a look at this PR ? If yes, could you approve so we can move forward with the conda release of the v1.1.2 of the `ruptures` package ?
> Many thanks for the time and energy !

@oboulant Done!
